code/modules/library/soapstone.dm

```diff
@@ -154,10 +154,27 @@ but only permanently removed with the curator's soapstone.
 
 /obj/structure/chisel_message/update_icon()
 	. = ..()
-	var/hash = md5(hidden_message)
-	var/newcolor = copytext_char(hash, 1, 7)
-	add_atom_colour("#[newcolor]", FIXED_COLOUR_PRIORITY)
-	set_light_color("#[newcolor]")
+
+	var/newcolor = COLOR_SOAPSTONE_BRONZE
+	switch(like_keys.len - dislike_keys.len)
+		if(-INFINITY to 0)
+			newcolor = COLOR_SOAPSTONE_IRON
+			name = "[initial(name)]"
+		if(1 to 10)
+			newcolor = COLOR_SOAPSTONE_BRONZE
+			name = "bronze [initial(name)]"
+		if(11 to 20)
+			newcolor = COLOR_SOAPSTONE_SILVER
+			name = "silver [initial(name)]"
+		if(21 to 30)
+			newcolor = COLOR_SOAPSTONE_GOLD
+			name = "gold [initial(name)]"
+		if(31 to INFINITY)
+			newcolor = COLOR_SOAPSTONE_DIAMOND
+			name = "diamond [initial(name)]"
```

Comment:
Feels a bit weird updating the `name` in `update_icon()`

I'd suggest to split it, but then you'd have to duplicate this switch logic twice. :shrug:

Reply:
we have update_appearance for this specifically.